motoman_gp4_support/urdf/gp4_macro.xacro

```diff
@@ -0,0 +1,159 @@
+    <link name="${prefix}link_3_u">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/visual/link_3_u.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_blue/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/collision/link_3_u.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_4_r">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/visual/link_4_r.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_blue/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/collision/link_4_r.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_5_b">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/visual/link_5_b.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_blue/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/collision/link_5_b.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+      <visual>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/visual/link_6_t.dae"/>
+        </geometry>
+        <xacro:material_yaskawa_silver/>
+      </visual>
+      <collision>
+        <geometry>
+          <mesh filename="package://motoman_gp4_support/meshes/gp4/collision/link_6_t.stl"/>
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}tool0"/>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+      <parent link="${prefix}base_link"/>
+      <child link="${prefix}link_1_s"/>
+      <origin xyz="0 0 0.330" rpy="0 0 0" />
+      <axis xyz="0 0 1" />
+      <limit lower="${radians(-170)}" upper="${radians(170)}" effort="220.5" velocity="${radians(465)}"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+      <parent link="${prefix}link_1_s"/>
+      <child link="${prefix}link_2_l"/>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <axis xyz="0 1 0" />
+      <limit lower="${radians(-110)}" upper="${radians(130)}" effort="305.76" velocity="${radians(465)}"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+      <parent link="${prefix}link_2_l"/>
+      <child link="${prefix}link_3_u"/>
+      <origin xyz="0 0 0.260" rpy="0 0 0" />
+      <axis xyz="0 -1 0" />
+      <limit lower="${radians(-65)}" upper="${radians(200)}" effort="95.55" velocity="${radians(525)}"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+      <parent link="${prefix}link_3_u"/>
+      <child link="${prefix}link_4_r"/>
+      <origin xyz="0.290 0 0.015" rpy="0 0 0" />
```

Comment:
This joint has translations in more than a single dimension.

We typically try to avoid that. One way to do that is to move the X translation to the next joint).

Reply:
This also makes the `origin` of `joint_4_r` coincident with `joint_5_b`. In most other models we don't do this.

Reply:
Actually, all our models are build this way.  GP8, MH12...  It is based on how Yaskawa builds there internal model.  So it makes it easier to create the model and the meshes orginal (imported from the 3D model) matches the joint_4_r coincident with the joint_5_b.

Reply:
Avoiding the translation in two dimensions has helped with generating things like IKFast plugins in the past.

Is that also from the DH parameters?

Reply:
Yes, it is from the HD parameters.
If you think that it will help with the integration with other tools, I can make the change and make it the standard for future models.

Reply:
Do the DH params then specify both a `d` and an `a` for that joint?

It has helped in the past, I don't know what the current state is of those code generators.

It would be good to find a compromise / workflow which works well for you, as you are the one who'd have to make manual changes all the time.

Reply:
Yes, the DH for that joint has both d and a values.  Making the change isn't a big amount of work.  What I'm more concern about is that (even if we don't currently publish the values), I believe that data for CG and Inertia of the links from our standard model is relative that joint location.  I'm not sure exactly how that normal converts to applications like Gazebo, but it might create problems down the line.